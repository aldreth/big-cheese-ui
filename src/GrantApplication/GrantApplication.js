import React from 'react';

const GrantApplication = ({grantApplication}) => (
  <tr className="stripe-dark">
    <td className="pa3">{grantApplication.name}</td>
    <td className="pa3">{grantApplication.postcode}</td>
    <td className="pa3">{grantApplication.amount}</td>
    <td className="pa3">{grantApplication.description}</td>
  </tr>
);

GrantApplication.propTypes = {
  grantApplication: React.PropTypes.object.isRequired
};

export default GrantApplication;
