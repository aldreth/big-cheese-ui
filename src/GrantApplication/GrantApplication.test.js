import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import GrantApplication from './GrantApplication';

const setQuestionsMock = jest.fn();

const grantApplication = {
  amount: '200',
  description: 'The cheese is amazing I make it with love and care. And I publicise it in the Daily Prophet.',
  name: 'Rita Skeeta',
  postcode: 'RS1 4RS',
};


const grantApplicationWrapper = shallow(<GrantApplication grantApplication={grantApplication} />);

it('renders amount', () => {
  expect(grantApplicationWrapper.text()).toContain('200');
});


it('renders description', () => {
  expect(grantApplicationWrapper.text()).toContain('The cheese is amazing I make it with love and care. And I publicise it in the Daily Prophet.');
});

it('renders name', () => {
  expect(grantApplicationWrapper.text()).toContain('Rita Skeeta');
});

it('renders postcode', () => {
  expect(grantApplicationWrapper.text()).toContain('RS1 4RS');
});

it('renders correctly', () => {
  const tree = renderer.create(
    <GrantApplication grantApplication={grantApplication} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
