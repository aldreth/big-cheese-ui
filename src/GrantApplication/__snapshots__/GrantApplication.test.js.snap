exports[`test renders correctly 1`] = `
<tr
  className="stripe-dark">
  <td
    className="pa3">
    Rita Skeeta
  </td>
  <td
    className="pa3">
    RS1 4RS
  </td>
  <td
    className="pa3">
    200
  </td>
  <td
    className="pa3">
    The cheese is amazing I make it with love and care. And I publicise it in the Daily Prophet.
  </td>
</tr>
`;
